pkg/util/listblocks/listblocks.go

```diff
@@ -60,28 +79,31 @@ func LoadMetaFilesAndDeletionMarkers(ctx context.Context, bkt objstore.BucketRea
 	})
 
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	if showDeleted {
-		deletionTimes, err = fetchDeletionTimes(ctx, bkt, deletionMarkerFiles)
+		deletionDetails, err = fetchMarkerDetails(ctx, bkt, DeletionMarker, deletionMarkerFiles)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, nil, err
 		}
 	}
-
+	noCompactBlocks, err = fetchMarkerDetails(ctx, bkt, NoCompactMarker, noCompactMarkerFiles)
+	if err != nil {
+		return nil, nil, nil, err
+	}
 	metas, err = fetchMetas(ctx, bkt, metaPaths)
-	return metas, deletionTimes, err
+	return metas, deletionDetails, noCompactBlocks, err
 }
 
 const concurrencyLimit = 32
 
-func fetchDeletionTimes(ctx context.Context, bkt objstore.BucketReader, deletionMarkers []string) (map[ulid.ULID]time.Time, error) {
+func fetchMarkerDetails(ctx context.Context, bkt objstore.BucketReader, markerType MarkerType, markers []string) (map[ulid.ULID]MarkerDetails, error) {
 	mu := sync.Mutex{}
-	times := map[ulid.ULID]time.Time{}
+	details := map[ulid.ULID]MarkerDetails{}
 
-	return times, concurrency.ForEachJob(ctx, len(deletionMarkers), concurrencyLimit, func(ctx context.Context, idx int) error {
-		r, err := bkt.Get(ctx, deletionMarkers[idx])
+	return details, concurrency.ForEachJob(ctx, len(markers), concurrencyLimit, func(ctx context.Context, idx int) error {
+		r, err := bkt.Get(ctx, markers[idx])
```

Comment:
the function was there, i didn't look into make it different because of the change size, it is not really relevant to the PR title. I see the difference between listblocks and other cases we use ReadMarker, is that here we do execute the read in concurrency. I would prefer not to change that in this PR, if possible it could be a good general refectory task for following.